src/components/molecules/ChatPoll/ChatPoll.tsx

```diff
@@ -0,0 +1,130 @@
+import React, { useCallback, useMemo } from "react";
+import classNames from "classnames";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faPoll } from "@fortawesome/free-solid-svg-icons";
+
+import { PollMessage, BaseMessageToDisplay, Vote } from "types/chat";
+
+import { WithId } from "utils/id";
+
+import { ChatMessageInfo } from "components/atoms/ChatMessageInfo";
+import { TextButton } from "components/atoms/TextButton";
+import Button from "components/atoms/Button";
+
+import { useUser } from "hooks/useUser";
+
+import "./ChatPoll.scss";
+
+export interface ChatPollProps {
+  pollData: WithId<BaseMessageToDisplay<PollMessage>>;
+  deletePoll: (pollId: string) => void;
+  voteInPoll: (votes: Vote[], pollId: string) => void;
+}
+
+export const ChatPoll: React.FC<ChatPollProps> = ({
+  pollData,
+  voteInPoll,
+  deletePoll,
+}) => {
+  const { user } = useUser();
+
+  const userId = user?.uid;
+
+  const { id, poll, votes, isMine } = pollData;
+  const { questions, topic, canBeDeleted } = poll;
+
+  const isVoted = userId
+    ? votes.map(({ userId }) => userId).includes(userId)
+    : false;
```

Comment:
this check should happen in the backend function

Reply:
If it is used for UI reasons here, it's fine for it to be here.

The backend function shouldn't need to use this check, as it should just be using `FieldValue.arrayUnion` to uniquely add the `userId` to the list of voting users. We shouldn't be storing an integer count in firestore at all. The count of votes will be the length of the array of 'voted `userId`s'.

Reply:
@0xdevalias agreed, if used for the UI reasons. The voting backend function should have the similar logic, 
>  using FieldValue.arrayUnion
is what I meant